verifier/contentsignature/verifier.go

```diff
@@ -62,25 +62,32 @@ func ParseChain(chain []byte) (certs []*x509.Certificate, err error) {
 // 3) has the x509v3 Extentions for CodeSigning use
 //
 // and SHA2 sum of raw bytes matches the provided hex-encoded with
-// optional colons rootHash param (as from openssl x509 -noout -text
+// optional colons rootHashes param (as from openssl x509 -noout -text
 // -fingerprint -sha256 -in ca.crt)
-func verifyRoot(rootHash string, cert *x509.Certificate) error {
+func verifyRoot(rootHashes []string, cert *x509.Certificate) error {
 	// this is the last cert, it should be self signed
 	if !bytes.Equal(cert.RawSubject, cert.RawIssuer) {
 		return fmt.Errorf("subject does not match issuer, should be equal")
 	}
 	if !cert.IsCA {
 		return fmt.Errorf("missing IS CA extension")
 	}
-	if rootHash == "" {
-		return fmt.Errorf("rootHash must not be empty")
+	if rootHashes == nil || len(rootHashes) < 1 {
+		return fmt.Errorf("rootHashes must not be empty")
 	}
-	rhash := strings.Replace(rootHash, ":", "", -1)
 	// We're configure to check the root hash matches expected value
 	h := sha256.Sum256(cert.Raw)
 	certHash := fmt.Sprintf("%X", h[:])
-	if rhash != certHash {
-		return fmt.Errorf("hash does not match expected root: expected=%s; got=%s", rhash, certHash)
+	var matchFound = false
+	for _, rootHash := range rootHashes {
```

Comment:
Refactored to check each root hash.
Argued with myself about removing colons ahead of time or in here. This seems like less code, the other way seems like less compute time. Minimal difference either way I think.

Reply:
You could use  [slices.ContainsFunc](https://pkg.go.dev/slices#ContainsFunc) or [slices.Contains](https://pkg.go.dev/slices#Contains) (if you move the `strings.Replace` to earlier) here.

Reply:
Ah yeah, `ContainsFunc` seems like the cleanest way.